plugins/errors_test.go

```diff
@@ -50,11 +54,83 @@ func TestInvalidConfigError(t *testing.T) {
 				PluginName: "caramlstore",
 			},
 			expected: "invalid extractor config",
+			hasError: false,
 		},
 	}
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
 			assert.Equal(t, tc.expected, tc.err.Error())
+			assert.Equal(t, tc.hasError, tc.err.HasError())
+		})
+	}
+}
+
+func TestNotFoundError(t *testing.T) {
+	type fields struct {
+		Type PluginType
+		Name string
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		expected string
+	}{
+		{
+			name: "WithType",
+			fields: fields{
+				Type: "extractor",
+				Name: "caramlstore",
+			},
+			expected: "could not find extractor \"caramlstore\"",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := NotFoundError{
+				Type: tt.fields.Type,
+				Name: tt.fields.Name,
+			}
+			if got := err.Error(); got != tt.expected {
+				t.Errorf("NotFoundError.Error() = %v, want %v", got, tt.expected)
+			}
```

Comment:
```suggestion
			assert.EqualError(t, err, tt.expected)
```

As mentioned early on, let's keep testing code consistent and use testify assertions along with referring to results as `expected` rather than `want`.